mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ar/ChessScene.kt

```diff
@@ -33,36 +40,47 @@ class ChessScene<Piece : ChessBoardState.Piece>(
     scope: CoroutineScope,
     startingBoard: Map<Position, Piece>,
 ) {
-  val boardNode: ArModelNode = ArModelNode(placementMode = PlacementMode.PLANE_HORIZONTAL)
 
-  private var currentPosition: Map<Position, Piece> = emptyMap()
-  private var currentPieces: MutableMap<Piece, ModelNode> = mutableMapOf()
+  /** The [ArModelNode] which acts as the root of the [ArModelNode] hierarchy. */
+  val boardNode = ArModelNode(placementMode = PlacementMode.PLANE_HORIZONTAL)
 
-  private var boardHeight: Float = 0f
-  private var boardHalfSize: Float = 0f
-
-  private var isLoaded = false
+  /** A conflated [Channel] which associates the position of the pieces to their values. */
+  private val currentPositionChannel =
+      Channel<Map<Position, Piece>>(capacity = CONFLATED).apply { trySend(startingBoard) }
 
   init {
     scope.launch {
-      // Load Board
-      val renderableInstance = loadBoard() ?: return@launch
-
-      // Once loaded compute the board size
-      val filamentAsset = renderableInstance.filamentAsset ?: return@launch
-      val boardBoundingBox = filamentAsset.boundingBox
-
-      // Get height (on y axe) of the board
-      // Double the value to get the total height of the box
-      boardHeight = 2 * boardBoundingBox.halfExtent[1]
-      boardHalfSize = boardBoundingBox.halfExtent[0]
-
-      // Load pieces
-      loadPieces(startingBoard)
-      isLoaded = true
 
-      // Move the pieces to the correct position
-      moveAllPiece()
+      // Load Board
+      val boardRenderableInstance = prepareBoardRenderableInstance(boardNode) ?: return@launch
+      val boundingBox = boardRenderableInstance.filamentAsset?.boundingBox ?: return@launch
+      val pieceRenderable = loadPieceRenderable()
+
+      currentPositionChannel
+          .consumeAsFlow()
+          .onEach { positions ->
+
+            // Remove all current children.
+            boardNode.children.forEach {
+              val child = boardNode.removeChild(it)
+              child.destroy()
+            }
```

Comment:
This seems a bad idea : 
- We cannot in the future add a move animation
- You shouldn't `destroy` node. A `renderableInstance` contains only data which is pass to the renderer engine and the engine create all the related assets (mesh, texture, material, etc...) Here, by calling `destroy` you free all the resources which means that when you call `setModel`, all the asserts will be created again. This is very expensive.